### ISSUE TITLE
Aspect ratio locked ignored if enableMouse()

### ISSUE THREAD

**Issue by leon-vv** [pyqtgraph/pyqtgraph#1346]
Thanks for the great project! This is such useful software!

### Short description

After calling `.aspectRatioLocked(True)` I was surprised to find that the aspect ratio was not kept constant. After debugging I found that directly calling .`enableMouse()` prevents the aspect ratio from being locked. This issue does not occur if `.setMouseEnabled()` is called. Maybe this is expected behavior, in that case consider this issue a request to better document the differences.

### Code to reproduce

```python
import numpy as np
import pyqtgraph as pg

data = np.random.normal(size=1000)

plt1 = pg.plot(np.arange(10), np.arange(10))
plt1.setAspectLocked(True)
plt1.enableMouse() # This prevents the asepect ratio to be locked
# plt1.setMouseEnabled() seems to work fine

if __name__ == '__main__':
    pg.QtGui.QApplication.exec_()
```

### Expected behavior
`.enableMouse()` and `.setMouseEnabled()` should not have an effect on whether the aspect ratio is locked or not.

### Real behavior
`.enableMouse()` overrules the locking of the aspect ratio.

### Tested environment(s)

 * PyQtGraph version: 0.11.0
 * Qt Python binding: 'PyQt5 5.15.0 Qt 5.15.0'
 * Python version: Python 3.7.6
 * NumPy version: 1.18.1
 * Operating system: Ubuntu
 * Installation method: Conda

**Comment by leon-vv** [comment]
As a workaround I found that listening to the signal directly on the graphics scene works without problems: `plotWidget.scene().sigMouseMoved.connect(self.plotWidgetMouseMove)`

**Comment by ixjlyons** [comment]
It's not quite clear to me what you're trying to achieve. Listen for mouse move events in the entire window? It does seem to be a little cumbersome with the way things are set up now (since `GraphicsView.enableMouse` enables scaling and dragging and whatnot which you may not want), so your workaround appears to be the way to go.

By the way, `plt1.setMouseEnabled()` eventually routes to the `ViewBox`, which has mouse enabled for both axes by default, so it doesn't really have an effect in the example above.

**Comment by leon-vv** [comment]
I simply wanted to listen on mouse move events on the displayed graph. I noticed the `sigSceneMouseMoved` signal on GraphicsView, which is only fired if `.enableMouse()` is called which in turn failed to keep the aspect ratio locked. So apparently I have to listen on the GraphicsScene instead of the GraphicsView? It might simply be ignorance on my part, but I found this very confusing.

**Comment by ixjlyons** [comment]
Right, I think the approach you've found is the best way to achieve what you want. See the [crosshair example](https://github.com/pyqtgraph/pyqtgraph/blob/master/examples/crosshair.py) if you're interested in how to map the scene coordinates to plot coordinates.

I agree this isn't really obvious. Enabling the mouse on the `GraphicsView` enables other kinds of mouse interactivity that may or may not be desirable. Perhaps it could always emit the signals but only act on the events if mouse is enabled. That could very well break other things though, I'm not sure at the moment.

The docs are long overdue for some descriptions of how to work with mouse and keyboard events etc. for various pyqtgraph objects. I could also imagine exposing events at the `PlotItem` level being more intuitive - I'd have to think about that some more.

Addressing the original issue though, I'm not sure I understand what you mean with the aspect ratio locking not working with `GraphicsView.enableMouse`. How are you checking this? A screenshot or two might be helpful. Setting `plt1.autoPixelRange = True` appears to make it respect the aspect ratio lock when resizing the window with mouse enabled - the x/y axes ranges change as the window is resized to keep the line at a 45-degree angle *on the computer screen*. Without it, the `GraphicsView` contents are distorted but the ranges are kept constant (so the aspect ratio is respected in plot coordinates). I'm not sure why `autoPixelRange` is set to False when `enableMouse` is called.

**Comment by leon-vv** [comment]
Hi @ixjlyons, thanks for your elaborate response.

> Setting plt1.autoPixelRange = True appears to make it respect the aspect ratio lock when resizing the window with mouse enabled - the x/y axes ranges change as the window is resized to keep the line at a 45-degree angle on the computer screen.

This is the desired behavior.

> Without it, the GraphicsView contents are distorted but the ranges are kept constant (so the aspect ratio is respected in plot coordinates).

Not sure what you mean by that. If the ranges wouldn't be kept constant the data shown wouldn't match with the data supplied to the library, that would be a major bug.

>  I'm not sure why autoPixelRange is set to False when enableMouse is called.

Exactly, that looks like a bug to me.

**Comment by HagaiHargil** [comment]
> The docs are long overdue for some descriptions of how to work with mouse and keyboard events etc. for various pyqtgraph objects. 

@ixjlyons Any news on that front? Specifically, I can't figure out how to bind keyboard events to buttons generated via a `ParameterTree` and a `ActionParameter` - the underlying widget is no where to be found.

Thanks!
